examples/lightning_base.py

```diff
@@ -60,8 +61,7 @@ def __init__(
         model=None,
         **config_kwargs
     ):
-        "Initialize a model."
-
+        """Initialize a model, tokenizer and config."""
         super().__init__()
         self.hparams = hparams
```

Comment:
we now also support more advanced ways of doing params

```
def __init__(self, lr, num_layers):
    super().__init_()
    self.save_hyperparameters()
     
     # this works
     Adam(self.hparams.lr)
```

And all your hparams are automatically saved now when you do the above.

althought self.hparams = ... still works. We just added more flexibility so you can pass in other models, losses, etc...